streaming/stream_blob.cc

```diff
@@ -44,15 +45,27 @@ static sstables::sstable_state sstable_state(const streaming::stream_blob_meta&
     return meta.sstable_state.value_or(sstables::sstable_state::normal);
 }
 
-static future<> load_sstable_for_tablet(const file_stream_id& ops_id, replica::database& db, table_id id, sstables::sstable_state state, sstables::entry_descriptor desc, seastar::shard_id shard) {
-    co_await db.container().invoke_on(shard, [id, desc, state, ops_id] (replica::database& db) -> future<> {
+static future<> load_sstable_for_tablet(const file_stream_id& ops_id, replica::database& db, db::view::view_building_worker& vbw, table_id id, sstables::sstable_state state, sstables::entry_descriptor desc, seastar::shard_id shard) {
+    auto& sharded_vbw = vbw.container();
+    co_await db.container().invoke_on(shard, [&sharded_vbw, id, desc, state, ops_id] (replica::database& db) -> future<> {
         replica::table& t = db.find_column_family(id);
         auto erm = t.get_effective_replication_map();
         auto& sstm = t.get_sstables_manager();
         auto sst = sstm.make_sstable(t.schema(), t.get_storage_options(), desc.generation, state, desc.version, desc.format);
         co_await sst->load(erm->get_sharder(*t.schema()));
         co_await t.add_sstable_and_update_cache(sst);
         blogger.info("stream_sstables[{}] Loaded sstable {} successfully", ops_id, sst->toc_filename());
+
+        if (state == sstables::sstable_state::staging) {
+            // If the sstable is in staging state, register it to view building worker
+            // to generate view updates from it.
+            // But because the tablet is still in migration process, register the sstable
+            // to the view building worker, which will create a view building task for it,
+            // so then, the view building coordinator can decide to process it once the migration
+            // is finished.
+            // (Instead of registering the sstable to view update generator which may process it immediately.)
+            co_await sharded_vbw.local().register_staging_sstable_tasks({sst}, t.shared_from_this());
+        }
```

Comment:
Off topic: I am not versed with this new building worker, but wouldn't it be better to just register the table and the worker pull staging sstables directly from the table? Looking at this interface only, it seems the worker might lose the opportunity to generate view updates from two sstables, streamed at different times, in a single round. I understand that the interface must keep backward compatibility with the old generator, so we might have no choice. But the worker might try to merge tasks, each with 1 sstable only, belonging to the same table, when migration finishes, right?